File: server/db/connection.js
const mongoose = require('mongoose');
require('dotenv').config();
const logger = require('../logger');
logger.info('MongoDB URI: %s', process.env.MONGODB_URI);
const url = process.env.MONGODB_URI;

mongoose.connect(url);

const db = mongoose.connection;

db.on('error', (err) => logger.error('connection error with mongodb: %s', err));
db.once('open', function () {
  logger.info('Connected successfully with mongoose');
  logger.info('Connection state with mongoose: %s', mongoose.connection.readyState);
});

File: go_server/db/connection.go
package db

import (
	"context"
	"log"
	"os"
	"time"

	"go.mongodb.org/mongo-driver/mongo"
	"go.mongodb.org/mongo-driver/mongo/options"
)

var Client *mongo.Client

func InitMongo() {

	uri := os.Getenv("MONGODB_URI")
	log.Printf("MongoDB URI: %s\n", uri)

	// Set client options
	clientOptions := options.Client().ApplyURI(uri)

	// Create context with timeout
	ctx, cancel := context.WithTimeout(context.Background(), 10*time.Second)
	defer cancel()

	// Connect to MongoDB
	client, err := mongo.Connect(ctx, clientOptions)
	if err != nil {
		log.Fatalf("Connection error with MongoDB: %v", err)
	}

	// Ping to verify connection
	err = client.Ping(ctx, nil)
	if err != nil {
		log.Fatalf("MongoDB ping failed: %v", err)
	}

	Client = client

	log.Println("Connected successfully with MongoDB")
}
